.github/workflows/docker-compose-tests.yml

```diff
@@ -1,5 +1,9 @@
 name: Docker Compose tests
 
+defaults:
+  run:
+    shell: bash --noprofile --norc -eo pipefail {0}
```

Comment:
IMHO pipefail should be set in bash script and not globally. AFAIK `set -euo pipefail` is considered a best practice in shell script writing.

Reply:
This is for the shell which github uses to execute a `run` command in GitHub workers. Doesn't have anything to do with bash scripts.

Reply:
Our test bash scripts have: `set -euf -o pipefail` so I am a bit confused why we need to set it again if Github action is just calling the scripts?

Reply:
When you have a github action like this:

```yaml
- name: Test ${{ matrix.test-setups.name }}
  env
    ...
  run: go test -race ./pkg/tests/end_to_end_tests/ ... | tee $SHORTNAME-test-run.log  2>&1
```

Github runs the `run` part in a shell. These parameters configure _that_ shell.

Reply:
This is the mechanism by which these errors were not being caught. The first command (the test) was failing, and piping the result into the log file, but because github's shell wasn't set to use `pipefail`, it was just silently accepting the failed tests.